freqtrade/optimize/recursive_analysis_helpers.py

```diff
@@ -0,0 +1,106 @@
+import logging
+import time
+from pathlib import Path
+from typing import Any, Dict, List
+
+from freqtrade.constants import Config
+from freqtrade.exceptions import OperationalException
+from freqtrade.optimize.recursive_analysis import RecursiveAnalysis
+from freqtrade.resolvers import StrategyResolver
+
+
+logger = logging.getLogger(__name__)
+
+
+class RecursiveAnalysisSubFunctions:
+
+    @staticmethod
+    def text_table_recursive_analysis_instances(
+            recursive_instances: List[RecursiveAnalysis]):
+        startups = recursive_instances[0]._startup_candle
+        headers = ['indicators']
+        for candle in startups:
+            headers.append(candle)
+
+        data = []
+        for inst in recursive_instances:
+            if len(inst.dict_recursive) > 0:
+                for indicator, values in inst.dict_recursive.items():
+                    temp_data = [indicator]
+                    for candle in startups:
+                        temp_data.append(values.get(int(candle), '-'))
+                    data.append(temp_data)
+
+        from tabulate import tabulate
+        table = tabulate(data, headers=headers, tablefmt="orgtbl")
+        print(table)
+        return table, headers, data
+
+    @staticmethod
+    def calculate_config_overrides(config: Config):
+        if 'timerange' not in config:
+            # setting a timerange is enforced here
+            raise OperationalException(
+                "Please set a timerange. "
+                "A timerange of 20 candles are enough for recursive analysis."
```

Comment:
interesting, this seems to contradict to what we're saying in the docs ... i think we need to adjust either this, or the docs (i think the docs are ... more correct in this regard).

> It is recommended to set a long timerange (at least 5000 candles) so that the initial indicators' calculation that is going to be used as a benchmark has very small or no recursive issues itself. For example, for a 5m timeframe, a timerange of 5000 candles would be equal to 18 days.

Reply:
Ah yes. On it rn